src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -186,14 +179,9 @@ object BoringUtils {
       case c: ComponentName => c.module
       case _ => throw new ChiselException("Can only add a Module or Component sink", null)
     }
-    val maybeDedup =
-      if (disableDedup) { Seq(new ChiselAnnotation { def toFirrtl = NoDedupAnnotation(moduleName) }) }
-      else { Seq[ChiselAnnotation]() }
-    val annotations =
-      Seq(new ChiselAnnotation {
-        def toFirrtl = SinkAnnotation(component.toNamed, name)
-      }) ++ maybeDedup
-    annotations.foreach(annotate(_))
+    // annotate doesn't support InstanceId (which is deprecated) because InstanceId doesn't implement toRelativeTarget
+    // this API is deprecated anyway so probably fine to not check it.
+    annotate()(Seq(SinkAnnotation(component.toNamed, name)) ++ Option.when(disableDedup)(NoDedupAnnotation(moduleName)))
```

Comment:
That seems okay to me. Also, given that this is deprecated since 6.0, delete in a follow-on? 💣

Reply:
That was exactly my thinking, I just figured it should be a separate PR